### PR TITLE
*: enable part revive for all code

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Problem Summary:

revive enable some rule to check all the tidb repo code.

the main problem is to fix the empty-line, If-Return and so on.

BTW, I upgrade the golangci-linter

### Check List

Tests 

- [x] Unit test

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by Yisaer — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]

File: ddl/db_rename_test.go [hunk 1]
```diff
@@ -155,20 +155,14 @@ func renameTableTest(t *testing.T, sql string, isAlterTable bool) {
 	failSQL := fmt.Sprintf(sql, "test_not_exist.t", "test_not_exist.t")
 	if isAlterTable {
 		tk.MustGetErrCode(failSQL, errno.ErrNoSuchTable)
-	} else {
-		tk.MustGetErrCode(failSQL, errno.ErrNoSuchTable)
```
Comment (xhebox) [inline ddl/db_rename_test.go:159]: Is logic here correct?

File: ddl/index.go [hunk 2]
```diff
@@ -426,7 +426,7 @@ func getNullColInfos(tblInfo *model.TableInfo, indexInfo *model.IndexInfo) ([]*m
 	return nullCols, nil
 }
 
-func checkPrimaryKeyNotNull(d *ddlCtx, w *worker, sqlMode mysql.SQLMode, t *meta.Meta, job *model.Job,
+func checkPrimaryKeyNotNull(d *ddlCtx, w *worker, _ mysql.SQLMode, t *meta.Meta, job *model.Job,
```
Comment (xhebox) [inline ddl/index.go:429]: Maybe just remove it.

File: ddl/index.go [hunk 3]
```diff
@@ -1008,7 +1008,7 @@ type addIndexWorker struct {
 	distinctCheckFlags []bool
 }
 
-func newAddIndexWorker(sessCtx sessionctx.Context, worker *worker, id int, t table.PhysicalTable, indexInfo *model.IndexInfo, decodeColMap map[int64]decoder.Column, reorgInfo *reorgInfo, jc *JobContext) *addIndexWorker {
+func newAddIndexWorker(sessCtx sessionctx.Context, _ *worker, id int, t table.PhysicalTable, indexInfo *model.IndexInfo, decodeColMap map[int64]decoder.Column, reorgInfo *reorgInfo, jc *JobContext) *addIndexWorker {
```
Comment (xhebox) [inline ddl/index.go:1011]: ditto.

File: ddl/index.go [hunk 4]
```diff
@@ -1456,7 +1456,7 @@ type cleanUpIndexWorker struct {
 	baseIndexWorker
 }
 
-func newCleanUpIndexWorker(sessCtx sessionctx.Context, worker *worker, id int, t table.PhysicalTable, decodeColMap map[int64]decoder.Column, reorgInfo *reorgInfo, jc *JobContext) *cleanUpIndexWorker {
+func newCleanUpIndexWorker(sessCtx sessionctx.Context, _ *worker, id int, t table.PhysicalTable, decodeColMap map[int64]decoder.Column, reorgInfo *reorgInfo, jc *JobContext) *cleanUpIndexWorker {
```
Comment (xhebox) [inline ddl/index.go:1459]: ditto

File: ddl/index.go (hunk 2 above)
Comment (hawkingrei) [reply ddl/index.go]: Done

File: ddl/index.go (hunk 4 above)
Comment (hawkingrei) [reply ddl/index.go]: Done

File: ddl/index.go (hunk 3 above)
Comment (hawkingrei) [reply ddl/index.go]: Done

**Review by qw4990 — APPROVED** [review]

File: ddl/db_rename_test.go (hunk 1 above)
Comment (hawkingrei) [reply ddl/db_rename_test.go]: fix

**Pull request merged**